ruby/README.md

```diff
@@ -16,8 +16,117 @@ Alternately, include "rcloadenv" in your bundle.
 You may then invoke the "rcloadenv" binary. You must pass the name of the
 runtime config resource, and then the command to execute. For example:
 
-    rcloadenv my-config -- bundle exec bin/rails s
+    bundle exec rcloadenv my-config -- bin/rails s
+
+If `rcloadenv` is run on Google Cloud Platform hosting (such as Google Compute
+Engine, Container Engine, or App Engine), then it infers the project name and
+credentials from the hosting environment.
+
+If you are not running on GCP, you may set the project using the `--project`
+switch, or by setting the `GOOGLE_CLOUD_PROJECT` environment variable. For
+example:
+
+    bundle exec rcloadenv --project=my-project my-config -- bin/rails s
+
+Run `rcloadenv --help` for more information on flags you can set.
+When not running on GCP, credentials are obtained from
+[Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials)
+so you can the `GOOGLE_APPLICATION_CREDENTIALS` environment variable or
+configure `gcloud auth`.
+
+## Example: Loading the Rails SECRET_KEY_BASE in Google App Engine
+
+This gem is commonly used to provide sensitive information such as API keys or
+database passwords to a Ruby application deployed to Google Cloud Platform,
+without exposing them to configuration files checked into source control.
+
+For example, Rails applications require the environment variable
+`SECRET_KEY_BASE` to be set in the production environment. When deploying a
+Rails application to Google App Engine, you could set this environment
+variable in the "app.yaml" configuration file, but that is not recommended
+because the "app.yaml" file is commonly checked into source control. Instead,
+you can set the `SECRET_KEY_BASE` value securely in the Runtime Config
+service, and use `rcloadenv` to load it into the Rails app. Here's how.
+
+1.  We will assume that you have a [Ruby on Rails](http://rubyonrails.org/)
+    application, you have set up a
+    [Google App Engine](https://cloud.google.com/appengine/) project to deploy
+    it to, you have the [Google Cloud SDK](https://cloud.google.com/sdk/)
+    installed, and you have logged in with gcloud and set your project name in
+    the gcloud configuration.
+
+    See https://cloud.google.com/ruby for more information on deploying a
+    Ruby application to Google App Engine.
+
+2.  Enable the Runtime Config API for your project using the Google Cloud
```

Comment:
I generally think it's better to show the numbering. It means you can read the document as text as well as compiled.